### PR TITLE
USHIFT-1052: Speed up greenboot check by checking Pod restarts in parallel

### PR DESCRIPTION
**Pull request by pmtk**

### CODE REVIEW

File: packaging/greenboot/microshift-running-check.sh [hunk 1]
```diff
@@ -119,9 +119,22 @@ for i in ${!PODS_NS_LIST[@]}; do
...
 
 # Verify that MicroShift core pods are not restarting
+declare -A pidNamespace
 for i in ${!PODS_NS_LIST[@]}; do
     CHECK_PODS_NS=${PODS_NS_LIST[$i]}
 
     echo "Checking pod restart count in the '${CHECK_PODS_NS}' namespace"
-    namespace_pods_not_restarting ${CHECK_PODS_NS}
+    namespace_pods_not_restarting ${CHECK_PODS_NS} &
+    pid=$!
```
Comment (ggiguash) [inline packaging/greenboot/microshift-running-check.sh:128]: Is there no race-condition here for the PID assignment? The function called in the background spawns many sub-shells, so how do we know that we're getting the "right" PID?

File: packaging/greenboot/microshift-running-check.sh [hunk 2]
```diff
@@ -119,9 +119,22 @@ for i in ${!PODS_NS_LIST[@]}; do
...
+    pidNamespace["${pid}"]="${CHECK_PODS_NS}"
+done
+
+set +e
+for pid in "${!pidNamespace[@]}"; do
+    wait "${pid}"
```
Comment (ggiguash) [inline packaging/greenboot/microshift-running-check.sh:134]: Can we just wait for all the sub-shells to exit? We should not need to disable exit-on-error - let's just fail the script if the background shell returns an error?

Comment (ggiguash) [reply packaging/greenboot/microshift-running-check.sh:134]: We should get an output about the failed background process, right?

**Review by dhellmann — COMMENTED** [review]
Since this is production code, it would be good to link the PR to a jira ticket

File: packaging/greenboot/microshift-running-check.sh (hunk 1 above)
Comment (pmtk) [reply packaging/greenboot/microshift-running-check.sh]: AFAIK `$!` is for most recent background process and we're not spawning any of those in `namespace_pods_not_restarting`. I did a small test which would suggest that we would be safe anyway:
```
+ pids=()
+ pids+=($!)
+ sleep 2
```

File: packaging/greenboot/microshift-running-check.sh (hunk 2 above)
Comment (pmtk) [reply packaging/greenboot/microshift-running-check.sh:132]: > We should get an output about the failed background process from bash, right?

You mean something like this?
```
[1]+  Exit 1 
```
I think it's only in interactive sessions

**Pull request merged**